spec/index.md

```diff
@@ -4670,33 +4680,66 @@ A record with
 
 -   `{certificate: C}`
 
-The predicate `may_read_path` is defined as follows, implementing the access control outlined in [Request: Read state](#http-read-state):
-
-    may_read_path(S, _, ["time"]) = True
-    may_read_path(S, _, ["subnet"]) = True
-    may_read_path(S, _, ["subnet", sid]) = True
-    may_read_path(S, _, ["subnet", sid, "public_key"]) = True
-    may_read_path(S, _, ["subnet", sid, "canister_ranges"]) = True
-    may_read_path(S, _, ["request_status", Rid]) =
-    may_read_path(S, _, ["request_status", Rid, "status"]) =
-    may_read_path(S, _, ["request_status", Rid, "reply"]) =
-    may_read_path(S, _, ["request_status", Rid, "reject_code"]) =
-    may_read_path(S, _, ["request_status", Rid, "reject_message"]) =
-    may_read_path(S, _, ["request_status", Rid, "error_code"]) =
+
+The predicate `may_read_path_for_canister` is defined as follows, implementing the access control outlined in [Request: Read state](#http-read-state):
+
+    may_read_path_for_canister(S, _, ["time"]) = True
+    may_read_path_for_canister(S, _, ["subnet"]) = True
+    may_read_path_for_canister(S, _, ["subnet", sid]) = True
+    may_read_path_for_canister(S, _, ["subnet", sid, "public_key"]) = True
+    may_read_path_for_canister(S, _, ["subnet", sid, "canister_ranges"]) = True
+    may_read_path_for_canister(S, _, ["subnet", sid, "metrics"]) = True
```

Comment:
@mraszyk I would like to get a confirmation on what we are expecting to do with allowing these under the canister endpoint. In the implementation, I would definitely still allow them for backwards compatibility (and allow end users to transition to the new endpoint) but I think we could also already remove them from the spec already.

The metrics one seems safe to only allow in the new endpoint given that it's a new path on the tree, so at the very least this one can be removed from here.

On a separate note: I've seen in the code that we also allow some paths of the form `["subnet", sid, "node", node_id])`, is this some other PR still in progress, so it doesn't appear here?

Reply:
> I would like to get a confirmation on what we are expecting to do with allowing these under the canister endpoint. In the implementation, I would definitely still allow them for backwards compatibility (and allow end users to transition to the new endpoint) but I think we could also already remove them from the spec as we are providing them through the new subnet endpoint.

I would keep them in the spec as long as we keep them in the actual implementation. If we want to deprecate them (by forbidding the paths in the HTTP handler), then I'd announce this intention as a note in the spec (at the moment we're just saying that the subnet form "should" be used and we can strengthen this statement in a further note).

> The metrics path seems safe to only allow in the new endpoint given that it's a new path on the tree, so at the very least this one can be removed from here.

Makes sense. Done

> On a separate note: I've seen in the code that we also allow some paths of the form ["subnet", sid, "node", node_id]), is this some other PR still in progress, so it doesn't appear here?

Indeed. Those paths are specified in this [PR](https://github.com/dfinity/interface-spec/pull/163) that could not be merged yet because the replica changes are still on hold due to performance benchmarks.

Reply:
> I would keep them in the spec as long as we keep them in the actual implementation. If we want to deprecate them (by forbidding the paths in the HTTP handler), then I'd announce this intention as a note in the spec (at the moment we're just saying that the subnet form "should" be used and we can strengthen this statement in a further note).

I'm fine with this, can you please edit that part in a way that it makes sense to you?

Reply:
I've added a deprecation announcement and a note. Please close this thread if the formulation makes sense to you.